File: hub/hub/schemas/questions.py
"""Question schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class QuestionCreate(BaseModel):
    from_agent: str
    question: str
    blocking: bool = False


class QuestionAnswer(BaseModel):
    answer: str


class QuestionResponse(BaseModel):
    id: str
    project_id: str
    from_agent: str
    question: str
    answer: Optional[str]
    answered: bool
    blocking: bool
    created_at: datetime
    answered_at: Optional[datetime]

    model_config = {"from_attributes": True}

File: hub/ui/src/api/questions.ts
import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
import { getJson, patchJson } from './client'
import { useConfigStore } from '@/store/configStore'

export interface Question {
  id: string
  project_id: string
  from_agent: string
  question: string
  blocking: boolean
  answered: boolean
  answer?: string
  created_at: string
  answered_at?: string
}

export function useQuestions(answered?: boolean) {
  const { isConfigured } = useConfigStore()
  const params = answered !== undefined ? `?answered=${answered}` : ''
  return useQuery<Question[]>({
    queryKey: ['questions', answered],
    queryFn: () => getJson<Question[]>(`/api/v1/questions${params}`),
    enabled: isConfigured,
  })
}

export function useAnswerQuestion() {
  const queryClient = useQueryClient()
  return useMutation({
    mutationFn: ({ id, answer }: { id: string; answer: string }) =>
      patchJson<Question>(`/api/v1/questions/${id}`, { answer }),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['questions'] })
      queryClient.invalidateQueries({ queryKey: ['status'] })
    },
  })
}
